hassio/src/addon-view/config/hassio-addon-config.ts

```diff
@@ -266,33 +266,43 @@ class HassioAddonConfig extends LitElement {
 
   private async _saveTapped(ev: CustomEvent): Promise<void> {
     const button = ev.currentTarget as any;
+    const eventdata = {
+      success: true,
+      response: undefined,
+      path: "options",
+    };
     button.progress = true;
 
     this._error = undefined;
 
     try {
-      await setHassioAddonOption(this.hass, this.addon.slug, {
+      const response = await setHassioAddonOption(this.hass, this.addon.slug, {
         options: this._yamlMode ? this._editor?.value : this._options,
       });
 
+      if (response.result === "error") {
+        this._error = this.supervisor.localize(
+          "addon.failed_to_save",
+          "error",
+          extractApiErrorMessage(response)
+        );
+        eventdata.success = false;
+      }
+
       this._configHasChanged = false;
-      const eventdata = {
-        success: true,
-        response: undefined,
-        path: "options",
-      };
-      fireEvent(this, "hass-api-called", eventdata);
       if (this.addon?.state === "started") {
         await suggestAddonRestart(this, this.hass, this.supervisor, this.addon);
       }
     } catch (err) {
```

Comment:
When does it throw, and when not on error? And why does it not always throw?

Reply:
It throws on request failure only

Reply:
Would it make sense to have `setHassioAddonOption` throw when `response.result === "error"`?

Reply:
Sure